scalameta/quasiquotes/shared/src/main/scala-3/scala/meta/quasiquotes/Api.scala

```diff
@@ -0,0 +1,196 @@
+    transparent inline def importee(inline args: Any*): Importee = ${ ReificationMacros.importeeImpl('sc, 'args) }
+    transparent inline def source(inline args: Any*): Source = ${ ReificationMacros.sourceImpl('sc, 'args) }
+  }
+
+  // unapply methods
+  extension (stringContext: scala.StringContext)
+    @annotation.compileTimeOnly(".q should not be called directly. Use q\"...\" string interpolation.")
+    def q: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".param should not be called directly. Use param\"...\" string interpolation.")
+    def param: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".t should not be called directly. Use t\"...\" string interpolation.")
+    def t: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".tparam should not be called directly. Use tparam\"...\" string interpolation.")
+    def tparam: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".p should not be called directly. Use p\"...\" string interpolation.")
+    def p: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".init should not be called directly. Use init\"...\" string interpolation.")
+    def init: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".self should not be called directly. Use self\"...\" string interpolation.")
+    def self: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".template should not be called directly. Use template\"...\" string interpolation.")
+    def template: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".mod should not be called directly. Use mod\"...\" string interpolation.")
+    def mod: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".enumerator should not be called directly. Use enumerator\"...\" string interpolation.")
+    def enumerator: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".importer should not be called directly. Use importer\"...\" string interpolation.")
+    def importer: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".importee should not be called directly. Use importee\"...\" string interpolation.")
+    def importee: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".source should not be called directly. Use source\"...\" string interpolation.")
+    def source: QuasiquoteUnapply = ???
+  
+  extension (inline sc: QuasiquoteUnapply)
+    transparent inline def unapply(scrutinee: Any): Any =
+      ${ ReificationMacros.unapplyImpl('sc, 'scrutinee) }
+
+
+  // parsers
+  object XTensionQuasiquoteTerm {
+    private[meta] def parse(input: scala.meta.inputs.Input, dialect: scala.meta.Dialect) = {
+      val parse = implicitly[Parse[Ctor]]
+      parse(input, dialect)
+    }.orElse{
+      val parse = implicitly[Parse[Stat]]
+      parse(input, dialect)
+    } match {
+      case x: scala.meta.parsers.Parsed.Success[_] => x.tree
+      case x: scala.meta.parsers.Parsed.Error => throw x.details
+    }
```

Comment:
isn't this a simple `.get` on the `Parsed` instance?

Reply:
Ah! That's true! This was one of the cases where I reused what was generated via macro annotation in scala 2, I missed this similarity completely. I'll try to simplify it.